File: src/main/java/work/xeltica/craft/core/events/StaffJoinEvent.java
package work.xeltica.craft.core.events;

import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;
import org.jetbrains.annotations.NotNull;

/**
 * スタッフがサーバーに参加したときに発生するイベント
 * @author Xeltica
 */
@SuppressWarnings("ALL")
public class StaffJoinEvent extends Event {
	@Override
	public @NotNull HandlerList getHandlers() {
        return HANDLERS_LIST;
	}

    public static HandlerList getHandlerList() {
        return HANDLERS_LIST;
    }

    private static final HandlerList HANDLERS_LIST = new HandlerList();
}

File: src/main/java/work/xeltica/craft/core/api/events/StaffJoinEvent.kt
package work.xeltica.craft.core.api.events

import org.bukkit.event.Event
import org.bukkit.event.HandlerList
import work.xeltica.craft.core.api.events.StaffJoinEvent

/**
 * スタッフがサーバーに参加したときに発生するイベント
 * @author Xeltica
 */
class StaffJoinEvent : Event() {
    override fun getHandlers(): HandlerList {
        return handlerList
    }

    companion object {
        val handlerList = HandlerList()
    }
}